### PR TITLE
Document the usage of our various date and datetime formats

### PR DESCRIPTION
**Pull request by TheDoubleK**
For this tech-debt ticket: https://trello.com/c/jyQZCEB7/303-review-datetime-filters

I've grepped through all our repos and I think everything here is true.  Honest.

### CODE REVIEW

File: dmutils/formats.py [hunk 1]
```diff
@@ -12,30 +12,59 @@
...
 
 
 def utcdatetimeformat(value, default_value=None):
     """
-    Use this filter to format deadline timestamps that are stored in the database as
+    Only use this filter to format deadline timestamps that are stored in the database as
     23:59:59 (UTC+00), when the localisation would otherwise display the date as 12.59am
-     on the next day (due to daylight savings), potentially causing confusion for buyers/suppliers.
+    on the next day (due to daylight savings), potentially causing confusion for buyers/suppliers.
+    This is currently (Jan 2018) used wherever we show the closing date for a brief.
```
Comment (katstevens) [inline dmutils/formats.py]: It's also used for the Clarification Questions deadline I think? (At least it should be!)

Comment (TheDoubleK) [reply dmutils/formats.py]: Yep, I can add that.

**Comment by TheDoubleK** [comment]
I'm not sure it is... the formats themselves are defined just above the formatter functions in the same file, but in theory these formats could be changed at any time and the formatters shouldn't care about it.

I guess the formats could have an example next to them, and then the examples would be more likely to be updated if the formats change.  I can add tomorrow.

**Comment by CrystalPea** [comment]
Good idea with adding the examples near the formats themselves, makes sense.

File: dmutils/formats.py [hunk 2]
```diff
@@ -47,6 +94,17 @@ def utcdatetimeformat(value, default_value=None):
 
 
 def datetodatetimeformat(value):
```
Comment (samuelhwilliams) [inline dmutils/formats.py]: Interesting. This method feels like it has the wrong name. It doesn't take a date (`date to`) and turn it into the same format as `datetimeformat`. It takes a date and turns it into the same format as comes out of our `dateformat`. Breaking change, but it feels like this should be called `datetodateformat`...

Comment (katstevens) [reply dmutils/formats.py]: We should also probably rename `utcdatetimeformat` to make it clear that we're *forcing* to UTC, but probably not in this PR?

**Comment by TheDoubleK** [comment]
This PR and associated tech debt ticket is all about **adding some comments** and definitely not about **breaking all the things**.

**Comment by TheDoubleK** [comment]
@CrystalPea I decided against having examples next to the formats themselves - I tried it out but I think it seems better with the formatter descriptions, and let's just hope people keep comments up to date if they change things in future.

File: dmutils/formats.py (hunk 2 above)
Comment (samuelhwilliams) [reply dmutils/formats.py]: Yeah, seems reasonable. But not for this PR as you say. 👍

**Pull request merged**